packages/mobile/src/fiatExchanges/ProviderOptionsScreen.tsx

```diff
@@ -105,100 +97,81 @@ function ProviderOptionsScreen({ route, navigation }: Props) {
-    onSelected: () => navigate(Screens.XanpoolScreen, providerWidgetInputs),
-  }
+    if (!isFocused) {
+      console.error(TAG, 'Screen is not in focus')
+      return
+    }
 
-  const moonpay = {
-    id: CicoProviderNames.Moonpay,
-    paymentMethods: [PaymentMethod.Card, PaymentMethod.Bank],
-    restricted: MOONPAY_RESTRICTED,
-    onSelected: () => navigate(Screens.MoonPayScreen, providerWidgetInputs),
-  }
+    if (!account) {
+      console.error(TAG, 'No account set')
+      return
+    }
 
-  const simplex = {
-    id: CicoProviderNames.Simplex,
-    paymentMethods: [PaymentMethod.Card],
-    restricted: SIMPLEX_RESTRICTED,
-    unavailable: !providerQuotes?.simplexQuote,
-    onSelected: () => {
-      if (providerQuotes?.simplexQuote && userLocation?.ipAddress) {
-        navigate(Screens.Simplex, {
-          simplexQuote: providerQuotes?.simplexQuote,
-          userIpAddress: userLocation.ipAddress,
-        })
-      }
-    },
-  }
+    if (!route.params.amount.fiat && !route.params.amount.crypto) {
+      console.error(TAG, 'No fiat or crypto purchase amount set')
+      return
+    }
 
-  const ramp = {
-    id: CicoProviderNames.Ramp,
-    paymentMethods: [PaymentMethod.Card, PaymentMethod.Bank],
-    restricted: RAMP_RESTRICTED,
-    onSelected: () => navigate(Screens.RampScreen, providerWidgetInputs),
-  }
+    try {
+      const providers = await fetchProviders({
+        userLocation,
+        walletAddress: account,
+        fiatCurrency: localCurrency,
+        digitalAsset: currencyToBuy,
+        fiatAmount: route.params.amount.fiat,
+        digitalAssetAmount: route.params.amount.crypto,
+      })
+      return providers
+    } catch (error) {
+      dispatch(showError(ErrorMessages.PROVIDER_FETCH_FAILED))
+    }
+  }, [userLocation, isFocused])
 
-  const transak = {
-    id: CicoProviderNames.Transak,
-    paymentMethods: [PaymentMethod.Card, PaymentMethod.Bank],
-    restricted: TRANSAK_RESTRICTED,
-    onSelected: () => navigate(Screens.TransakScreen, providerWidgetInputs),
-  }
+  const activeProviders = asyncProviders.result
 
-  const providers: {
+  const cicoProviders: {
     cashOut: CicoProvider[]
     cashIn: CicoProvider[]
   } = {
-    cashOut: [xanpool].sort(sortProviders),
-    cashIn: [moonpay, simplex, xanpool, ramp, transak].sort(sortProviders),
+    cashOut: activeProviders?.filter((provider) => provider.cashOut).sort(sortProviders) || [],
+    cashIn: activeProviders?.filter((provider) => provider.cashIn).sort(sortProviders) || [],
   }
 
   const providerOnPress = (provider: CicoProvider) => () => {
+    if (provider.unavailable) {
+      return
+    }
+
     ValoraAnalytics.track(FiatExchangeEvents.provider_chosen, {
       isCashIn,
-      provider: provider.id,
+      provider: provider.name,
     })
-    dispatch(selectProvider(provider.id))
-    provider.onSelected()
+
+    dispatch(selectProvider(provider.name))
+
+    if (provider.name === 'Simplex') {
```

Comment:
Eventually we might want to have an enum for integrated providers, but this is fine for now.

Reply:
Done.